src/griddyn/Generator.h

```diff
@@ -27,7 +29,7 @@ class gridSubModel;
 gridSecondary class and inherits from that class it has mechanics for dealing with the power flow solution modes of
 a generator with no dynamics
 */
-class Generator : public gridSecondary
+class Generator : public gridSecondary, public griddyn_generator_
```

Comment:
I am not sure I like having the multiple inheritance here, especially with a 1 byte size.  That is going to screw with the memory alignment and make for some rather complicated objects.  Is there any way we can Not do this?

Reply:
I'll double check that the structs need to be defined on the C-side, rather than just declared. That would let them be empty, at least.

The goal is to return pointers to griddyn things as pointers to real C things in a type safe way. That means no pointers to void. The only way I know of doing this is with a common base object between C/C++, but which isn't common between the C things.

Reply:
As they're now empty, the empty base optimization should kick in. Memory alignment should be as before, this should only be visible on the type level.